services/QuillLMS/app/controllers/teachers/unit_activities_controller.rb

```
@@ -23,10 +23,8 @@ def hide
     render json: {}
   end
 
-  def update_multiple_due_dates
-    base_unit_activities = UnitActivity.where(id: params[:unit_activity_ids])
-    activity_ids = base_unit_activities.map(&:activity_id)
-    UnitActivity.where(activity_id: activity_ids, unit_id: base_unit_activities.first.unit_id).update_all(due_date: params[:due_date])
+  def update_multiple_dates
+    UnitActivity.where(id: params[:unit_activity_ids]).each { |ua| ua.update(params[:date_attribute] => params[:date]) }
```

I would add a check here that 
```
if params[:date_attribute].in?('publish_date', 'due_date')
  UnitActivity
    .where(id: params[:unit_activity_ids])
    .each { |ua| ua.update(params[:date_attribute] => params[:date]) }
end
```
 to ensure that only the columns we're expecting to be updated are updated.

For example, we wouldn't want this to be allowed: `put :update_multiple_dates, params: { unit_activity_ids: [unit_activity.id, unit_activity2.id, unit_activity3.id], date_attribute: 'activity_id', date: 5 }`

---

I think there's also a larger question about error states here.  Currently, if the update fails, it fails silently.  For that reason I tend to prefer `update!` over `update`, in which case you'll need a transaction block since there are multiple objects being created:

```
ActiveRecord::Base.transaction do
  UnitActivity
      .where(id: params[:unit_activity_ids])
      .each { |ua| ua.update!(params[:date_attribute] => params[:date]) }
end
```